### PR TITLE
Read-only variant details

### PR DESCRIPTION
**Pull request by 0nko**
This PR adds read-only product details screen. I'll fix the failing tests in a separate PR.

| Light mode | Dark mode|
| - | - |
| image | image |
| image | image 
| image | image |

**To test:**

1. Open a variable product from the product list
2. Notice it shows a property group with attribute names and counts
3. Tap on Variations
4. A list of variations is loaded
5. Tap on one with an image
6. Notice a variation detail is loaded, the title is in the toolbar and it shows visibility, description (if not empty), price (including sale price or scheduled sale) and product type
7. Tap on the image
8. Notice the image detail is displayed

### CODE REVIEW

**Comment by 0nko** [comment]
Thanks for the review, @Garance91540!

> Hi @0nko , I downloaded the APK file but still wasn't able to go through the flow (I was basically stuck at this screen). Is it the right file?

Yes, sorry. There was a feature flag that enabled the tap only for debug builds. It should work now.

> This is the design of a variant product page, is it possible to update it? It includes Parent product images, title, description, product type per default.

Yes, definitely. I was just planning to do it in a separate PR because this one getting too big as it is.

> In order to align with core product types (simple, variable, external, grouped), variations are always "variable products", so we can remove the "Product type" row since it should appear on the parent product screen

:+1:

> *The "visible" row looks disable to me. Is it possible to use the font Subtitle 1, color "On Surface / High Emphasis", and to add the toggle icon to turn it on and off? Here is the design

:+1:

> I'm assuming we'll add the "Inventory" and "Shipping" rows on the Variant product screen in later PR, just want to remind it here so we don't forget (maybe we should create issues now? I'll leave it up to you!)

Yes. Thanks, I'll create those issues :+1:.

> PS: If you have the chance in the future to share some before/after screenshots and a quick screencast recording of the new flow, this can drastically improve my response time 🙂

:+1:

By the way, you probably noticed the variant image fills the full width of the screen. It's because variants can only have 1 image, as opposed to products. So I removed the gallery view and made the single image clickable, which makes it full-screen.

**Comment by Garance91540** [comment]
> This is the design of a variant product page, is it possible to update it? It includes Parent product images, title, description, product type per default.

Yes, definitely. I was just planning to do it in a separate PR because this one getting too big as it is.

👍

> By the way, you probably noticed the variant image fills the full width of the screen. It's because variants can only have 1 image, as opposed to products. So I removed the gallery view and made the single image clickable, which makes it full-screen.

👍

**Comment by 0nko** [comment]
@Garance91540 I've created issues for the missing shipping and inventory rows and also the product detail changes. I'll work on those next.

In this PR I've updated the visibility row to be a switch (which is currently disabled) and it's using the Subtitle1 style.

Here's a short demo:

variation_details

**Comment by 0nko** [comment]
> Hi @0nko looks good the only thing is that the "Visible" toggle is activated and so should be purple

@Garance91540 It's like this on purpose since the screen is "read-only" for now. It'll be activated once you can edit the variant.

File: WooCommerce/src/main/res/layout/product_property_switch_layout.xml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        style="@style/Woo.Divider"
+        android:layout_width="@dimen/minor_00"
+        android:layout_marginStart="@dimen/major_200"
+        android:layout_marginTop="@dimen/major_100"
+        app:layout_constraintBottom_toBottomOf="parent"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toEndOf="@+id/switchIcon"
+        app:layout_goneMarginStart="@dimen/major_100" />
+
+</androidx.constraintlayout.widget.ConstraintLayout>
```
Comment (AmandaRiu) [inline WooCommerce/src/main/res/layout/product_property_switch_layout.xml:60]: np: this works fine, but it's definitely a component that gets resused across the app - we even have a generic version of it already: `WCSettingsToggleOptionView` - what do you think about using that instead to cut down on code duplication?

File: WooCommerce/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -58,6 +58,7 @@
     <string name="products">Products</string>
     <string name="refunds">Refunds</string>
     <string name="product_variations">Variations</string>
+    <string name="product_variation_options">%s (%s options)</string>
```
Comment (AmandaRiu) [inline WooCommerce/src/main/res/values/strings.xml:61]: np: Should this use numbered string placeholders? ie: `%1$s (%2$s options)`

Comment (0nko) [reply WooCommerce/src/main/res/values/strings.xml:61]: Oh yeah, good call!

File: WooCommerce/src/main/res/layout/product_property_switch_layout.xml (hunk 1 above)
Comment (0nko) [reply WooCommerce/src/main/res/layout/product_property_switch_layout.xml]: I've tried to reuse the `WCSettingsToggleOptionView` as you suggested but it turned out to be more nuisance than it's worth. The variation property switch property doesn't have a description but it needs an extra divider (which the settings toggle layout doesn't have). When I added the missing views, I had to add logic to hide/show them in both places. Also, the horizontal paddings are a bit different in the preferences screen so overall, even though it looks like the view could be reused, it just resulted in more unreadable code, so I got rid of it.

**Comment by 0nko** [comment]
@AmandaRiu Thanks for the review! Ready for another round.

**Pull request merged**